### PR TITLE
Refactor the code

### PR DESCRIPTION
**Pull request by hafiz1379**
### Summary
This PR focuses on codebase cleanup and improving the structure of the application to adhere to best practices, including YAGNI (You Aren't Gonna Need It), KISS (Keep It Simple, Stupid), DRY (Don't Repeat Yourself), and the Ruby style guide. Additionally, SOLID principles, with a special emphasis on the Single-responsibility Principle, have been followed to create a cleaner and more maintainable codebase.

### Changes Made

**Codebase Cleanup**
- Conducted a thorough code review and identified areas for improvement.
- Removed unnecessary or redundant code that violated the YAGNI and DRY principles.
- Simplified complex logic to adhere to the KISS principle.

**Application Structure Enhancement**
- Extracted and refactored pieces of logic into separate classes to promote single responsibility and better code organization.
- Introduced classes for managing books, people, and rentals, ensuring each class has a clear and distinct purpose.
- Created a separate class, `Menu`, for handling user interactions and menu options.
- Updated `main.rb` to serve solely as the entry point for the application, responsible for starting the code and asking the user for initial choices or quitting.

**Style Guide and SOLID Principles**
- Ensured the codebase follows the Ruby style guide for readability and consistency.
- Adhered to SOLID principles, with particular emphasis on the Single-responsibility Principle, to create more maintainable and modular code.

### Reviewer's Notes
I invite the code reviewer to thoroughly examine the refactored codebase and provide feedback on code quality, design, adherence to best practices, and adherence to SOLID principles. Your insights are highly valued. Thank you for taking the time to review this PR.

### CODE REVIEW

File: Lib/app_interface.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  def handle_option(option)
+    action = ACTIONS[option]
+    if action
+      @app.send(action)
+    else
+      puts 'Invalid option. Please try again.'
+    end
+  end
+end
```
Comment (V-Blaze) [inline Lib/app_interface.rb:110]: - You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :100: 
- [ ] In your `app.rb`, in the `create_student and create_teacher ` method, all requests for input and input validation should've been placed in different classes/methods whose responsibilities are to request or validate input. Following the `single responsible rule`, the methods don't need to handle, user input collection, validation, and creating the student or teacher object. I'll suggest you update this. :+1: :100: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @hafiz1379, @JuanDiegoGuerra  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- App works as expected✔
- Correct git flow👋 
-  Readme looks professional✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @hafiz1379 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby features ✔️  
- [x] Kudus on meeting the requirements  as suggested by the last reviewer ✔️


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by hafiz1379** [comment]
Thank you @Shedrack-Sunday

**Pull request merged**